codice/saml-conformance
# MAINT - Improved Reporting for CORE

### Description of the Change
- Added the ability to have a report and continue running tests when an error occurs
- Used a Report object (singleton) to add errors instead of throwing exceptions
- Added the ability to run quietly which is just the report without the errors
- Added a codes field in SAMLComplianceException and the ability to compare them

**Hopefully, the separate commits help.**
Updates:
- I have location-wise moved the report package to be near the SAMLComplainceException one
- Now both a SAMLRefMessage and a SAMLComplainceException have knowledge of what section the exception corresponds to so no need of passing section around.

### SAML V2.0 Specification Quote
N/A

### Verification Process
- Make sure that failures are reported successfully 
- Make sure tests are skipped successfully 
- Make sure DDF passes

### Screenshots
## **NOT** quiet mode:
![screen shot 2019-01-03 at 10 43 10 am](https://user-images.githubusercontent.com/29550062/50655688-1d014900-0f4e-11e9-9d8e-dd89955b55b8.png)
![screen shot 2019-01-03 at 10 43 25 am](https://user-images.githubusercontent.com/29550062/50655689-1d014900-0f4e-11e9-8164-1872012e80aa.png)

## Quiet mode:
<img width="1107" alt="screen shot 2019-01-03 at 11 20 52 am" src="https://user-images.githubusercontent.com/29550062/50655728-37d3bd80-0f4e-11e9-95d6-1586979ae7bc.png">
<img width="1101" alt="screen shot 2019-01-03 at 11 23 22 am" src="https://user-images.githubusercontent.com/29550062/50655729-37d3bd80-0f4e-11e9-8ee2-f770cb832a32.png">

### Checklist:
